lightly_studio/tests/core/test_dataset.py

```diff
@@ -356,6 +356,108 @@ def test_compute_typicality_metadata(
         assert samples[1].metadata["typicality"] == pytest.approx(0.4222289)
         assert samples[2].metadata["typicality"] == pytest.approx(0.3853082)
 
+    def test_add_samples_from_path_tag_depth_invalid(
+        self,
+        patch_dataset: None,  # noqa: ARG002
+    ) -> None:
+        """Tests that tag_depth > 1 raises an error."""
+        dataset = Dataset.create(name="test_dataset")
+        with pytest.raises(
+            NotImplementedError,
+            match="tag_depth > 1 is not yet implemented for add_samples_from_path",
+        ):
+            dataset.add_samples_from_path(".", tag_depth=2)
```

Comment:
All 3 tests currently call `dataset.add_samples_from_path` and not `_tag_samples_by_directory`. Thus they are more integration tests than unittests.

I would change them to call `_tag_samples_by_directory` instead. 
From a testing point of view, this makes sure that only and exactly the relevant part is tested.
From a test implementation point of view, it allows to leave away all the the mocking of `lightly_studio.dataset.fsspec_lister.iter_files_from_path`, the files do not need to be on disk anymore. It is sufficient if the samples are in the DB, which happens with `create_images(`.